core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2792,12 +2792,14 @@ protected Scope visitMerge(Merge merge, Optional<Scope> scope)
         {
             Relation relation = merge.getTarget();
             Table table = getMergeTargetTable(relation);
-            QualifiedObjectName tableName = createQualifiedObjectName(session, table, table.getName());
-            if (metadata.getView(session, tableName).isPresent()) {
+            QualifiedObjectName originalTableName = createQualifiedObjectName(session, table, table.getName());
+            if (metadata.getView(session, originalTableName).isPresent()) {
```

Comment:
pre-existing q: do we need a `getMaterializedView` check here as well OR it's supported for merge somehow? AFAICT we exclude MVs for listing/getting views

Reply:
cc @djsstarburst

Reply:
Do note that `visitDelete` and `visitUpdate` methods from `io.trino.sql.analyzer.StatementAnalyzer.Visitor` don't check for MVs as well even though AFAIK such operations are not supported on MVs. I can create a separate PR to cope with this issue.

cc: @findepi

Reply:
MV cannot be written to.
A query attempting to INSERT into /  UPDATE / DELETE from / MERGE into a MV should fail cleanly.

Reply:
Created https://github.com/trinodb/trino/pull/14062 follow-up PR